src/main/java/uk/gov/ons/census/casesvc/messaging/SampleReceiver.java

```diff
@@ -9,75 +12,124 @@
 import uk.gov.ons.census.casesvc.model.dto.Payload;
 import uk.gov.ons.census.casesvc.model.entity.Case;
-import uk.gov.ons.census.casesvc.model.entity.CaseStatus;
+import uk.gov.ons.census.casesvc.model.entity.CaseState;
+import uk.gov.ons.census.casesvc.model.entity.UacQidLink;
 import uk.gov.ons.census.casesvc.model.repository.CaseRepository;
+import uk.gov.ons.census.casesvc.model.repository.EventRepository;
+import uk.gov.ons.census.casesvc.model.repository.UacQidLinkRepository;
+import uk.gov.ons.census.casesvc.utility.IacDispenser;
+import uk.gov.ons.census.casesvc.utility.QidCreator;
 
 @MessageEndpoint
 public class SampleReceiver {
   private CaseRepository caseRepository;
+  private UacQidLinkRepository uacQidLinkRepository;
+  private EventRepository eventRepository;
   private RabbitTemplate rabbitTemplate;
+  private IacDispenser iacDispenser;
+  private QidCreator qidCreator;
+  private MapperFacade mapperFacade;
 
   @Value("${queueconfig.emit-case-event-exchange}")
   private String emitCaseEventExchange;
 
-  public SampleReceiver(CaseRepository caseRepository, RabbitTemplate rabbitTemplate) {
+  public SampleReceiver(
+      CaseRepository caseRepository,
+      UacQidLinkRepository uacQidLinkRepository,
+      EventRepository eventRepository,
+      RabbitTemplate rabbitTemplate,
+      IacDispenser iacDispenser,
+      QidCreator qidCreator,
+      MapperFacade mapperFacade) {
     this.caseRepository = caseRepository;
     this.rabbitTemplate = rabbitTemplate;
+    this.iacDispenser = iacDispenser;
+    this.uacQidLinkRepository = uacQidLinkRepository;
+    this.eventRepository = eventRepository;
+    this.qidCreator = qidCreator;
+    this.mapperFacade = mapperFacade;
   }
 
   @Transactional
-  @ServiceActivator(inputChannel = "amqpInputChannel")
-  public void receiveMessage(FooBar fooBar) {
-    System.out.println(fooBar.getFoo());
-    Case caze = new Case();
-    caze.setId(UUID.randomUUID());
-    caze.setStuff(fooBar.getFoo());
-    caze.setStatus(CaseStatus.NOTSTARTED);
-    caseRepository.save(caze);
+  @ServiceActivator(inputChannel = "caseSampleInputChannel")
+  public void receiveMessage(CreateCaseSample createCaseSample) {
+
+    Case caze = persistToDatabase(createCaseSample);
+
+    CaseCreatedEvent caseCreatedEvent = prepareEventToEmit(caze);
+
+    rabbitTemplate.convertAndSend(emitCaseEventExchange, "", caseCreatedEvent);
+  }
+
+  private Case persistToDatabase(CreateCaseSample createCaseSample) {
+    Case caze = mapperFacade.map(createCaseSample, Case.class);
+    caze.setCaseId(UUID.randomUUID());
+    caze.setState(CaseState.ACTIONABLE);
+    caze = caseRepository.saveAndFlush(caze);
+
+    UacQidLink uacQidLink = new UacQidLink();
+    uacQidLink.setId(UUID.randomUUID());
+    uacQidLink.setUac(iacDispenser.getIacCode());
+    uacQidLink.setCaze(caze);
+    uacQidLink = uacQidLinkRepository.saveAndFlush(uacQidLink);
+
+    int questionnaireType = 99;
+    int trancheIdentifier = 2;
```

Comment:
We should make this a configurable value in `application.yml` so that we can change the number between 2019 and 2021 census.